src/radio.c

```diff
@@ -565,5 +565,30 @@ void handle_radio_request(uint8_t request, uint8_t * data)
     rfce = 1;
     in1bc = 1;
   }
+
+  else if (request == SPI_TRANSACTION) {
+    const bool assert_csn   = (data[0] & 0x80) != 0;
+    const bool deassert_csn = (data[0] & 0x40) != 0;
+    const uint8_t len       = (data[0] & 0x1f);
+    uint8_t i = 0;
+
+    if (assert_csn) {
+        platform_assert_spi_master_cs(true);
+    }
+
+    if (len != 0) {
+        for (i = 0; i < len; i++)
+        {
+            SMDAT = data[i+1];
+            in1buf[i] = SMDAT;
```

Comment:
I need to apply a fix here.

SMDAT should not be read until the MSDONE interrupt has fired.  In dumping nRF24LU/E1+ flash, I've observed bad reads.

A quick hack is to add a delay_us() between the write and read of SMDAT.  However, I'll look into a proper ISR and flag.  (Grr... I wish SMCTRL just had a darn status bit...)